packages/next/src/build/utils.ts

```diff
@@ -1479,27 +1486,41 @@ export async function buildAppStaticPaths({
         }
 
         const builtParams = await buildParams()
-        const fallback = !generateParams.some(
+
+        if (
+          generateParams.some(
+            (generate) => generate.config?.dynamicParams === true
+          ) &&
```

Comment:
The checks are different here, one is checking if any is false, and one if any is true. We can't just invert it because it may not be defined at all too.